--- a/luci-app-mosdns/root/etc/init.d/mosdns
+++ b/luci-app-mosdns/root/etc/init.d/mosdns
@@ -6,7 +6,6 @@
 USE_PROCD=1
 
 PROG=/usr/bin/mosdns
-CONF=$(uci -q get mosdns.config.configfile)
 CRON_FILE=/etc/crontabs/root
 DUMP_FILE=/etc/mosdns/cache.dump
 DUMP_FILE_DEFAULT=/usr/share/mosdns/cache.dump
@@ -62,6 +61,7 @@
 	config_get stats_capacity $1 stats_capacity 2000
 	config_get stats_dump_file $1 stats_dump_file 0
 	config_get stats_dump_interval $1 stats_dump_interval 600
+	config_get CONF $1 configfile "/var/etc/mosdns.json"
 }
 
 generate_config() {
@@ -814,8 +814,6 @@
 
 	rm -rf /tmp/log/mosdns*
 
-	$MOSDNS_SCRIPT v2dat_dump
-
 	[ "${CONF}" = "/var/etc/mosdns.json" ] && generate_config
 
 	procd_open_instance mosdns
